### DISCUSSION TITLE
Linux-ft-fms-docker and hosting a web-page using WebDirect htdocs / httpsRoot

### DISCUSSION THREAD

**Discussion by ayescas** [fmgarage/ft-fms-docker#8] [Q&A]
Hello,

Thank you so much for ft-fms-docker.  I use it every day and love it.
Question is it possible to host a home-page using WebDirect?

I created these two folders in the "HTTPServer" directory:

htdocs / httpsRoot / index.html

Every time I go to localhost, I keep getting the FileMaker Webdirect test page.  

Thank you.
Alex

**Comment by sk-gara** [comment]
Hello Alex,

sorry for letting you wait so long, but I did not receive any notification from github about your post.
It is really nice to hear that you find this tool useful, we appreciate it!

As for your question, I assume that you create the htdocs/... folder on the host in the fms-data folder, is that right?
This will not be mounted inside the container, the only mountpoints inside HTTPServer are .conf, conf and logs.
We could add htdocs, if there is a plausible usecase for it.

So, what benefit will hosting a custom webpage with webdirect provide?

Greetings,
Steffen

**Comment by ayescas** [comment]
Hey Steffen,

Thank you for the reply.  In our case, we host our website from the htdocs folder.  All our dynamic forms use web-direct and are embedded to the HTML pages using iframes.  It would be awesome if we could load our webpage using localhost.  With the current set-up, we can only view the FileMaker Web-Direct structure and not the whole website, which contains the iframes and web-direct.

I was able to view the htdocs folder in VSC and created a folder called httpsRoot, then copied my website to this folder, but when I go to localhost, I get the "FileMaker Database Server Website" default page instead of index.html.

On Windows server, our website is hosted on FileMakerServer / HTTPServer / Conf / index.html

Thank you so much, this tool is valuable to my local development.

-Alex


From: S ***@***.***>
Date: Thursday, July 29, 2021 at 2:34 AM
To: fmgarage/ft-fms-docker ***@***.***>
Cc: Alejandro Yescas ***@***.***>, Author ***@***.***>
Subject: Re: [fmgarage/ft-fms-docker] Linux-ft-fms-docker and hosting a web-page using WebDirect htdocs / httpsRoot (#8)

Hello Alex,

sorry for letting you wait so long, but I did not receive any notification from github about your post.
It is really nice to hear that you find this tool useful, we appreciate it!

As for your question, I assume that you create the htdocs/... folder on the host in the fms-data folder, is that right?
This will not be mounted inside the container, the only mountpoints inside HTTPServer are .conf, conf and logs.
We could add htdocs, if there is a plausible usecase for it.

So, what benefit will hosting a custom webpage with webdirect provide?

Greetings,
Steffen

**Comment by sk-gara** [comment]
Alright, I'll make this an issue and add htdocs to the externally available volumes.

**Comment by sk-gara** [comment]
I just pushed a commit that includes htdocs and did a quick test that worked, please let me know if that works for you.

**Comment by ayescas** [comment]
Omg… Thank you so much. Can’t wait to use it.

**Comment by ayescas** [comment]
I added the httpsRoot folder in the htdocs directory and when I go to localhost I still get the default FileMaker page instead of loading my website.  Any ideas?

  *   Linux for HTTPS: /opt/FileMaker/FileMaker Server/HTTPServer/htdocs/httpsRoot
  *
***@***.***

From: Alejandro Yescas ***@***.***>
Date: Friday, July 30, 2021 at 6:52 AM
To: fmgarage/ft-fms-docker ***@***.***>
Cc: fmgarage/ft-fms-docker ***@***.***>, Author ***@***.***>
Subject: Re: [fmgarage/ft-fms-docker] Linux-ft-fms-docker and hosting a web-page using WebDirect htdocs / httpsRoot (#8)
Omg… Thank you so much. Can’t wait to use it.


On Jul 30, 2021, at 3:27 AM, S ***@***.***> wrote:
﻿

I just pushed a commit that includes htdocs and did a quick test that worked, please let me know if that works for you.

**Comment by ayescas** [comment]
Please disregard my previous message.

I did a clean reinstall of ft-fms-docker and it works perfectly !!!

-Alex



From: S ***@***.***>
Date: Friday, July 30, 2021 at 3:27 AM
To: fmgarage/ft-fms-docker ***@***.***>
Cc: Alejandro Yescas ***@***.***>, Author ***@***.***>
Subject: Re: [fmgarage/ft-fms-docker] Linux-ft-fms-docker and hosting a web-page using WebDirect htdocs / httpsRoot (#8)

I just pushed a commit that includes htdocs and did a quick test that worked, please let me know if that works for you.

**Comment by ayescas** [comment]
The httpd.conf has PHP disabled.  Is it possible to enable it, if so do we just delete the following:

#
# Disable PHP document
#
<FilesMatch "\.php$">
    Require all denied
</FilesMatch>



From: S ***@***.***>
Date: Friday, July 30, 2021 at 3:27 AM
To: fmgarage/ft-fms-docker ***@***.***>
Cc: Alejandro Yescas ***@***.***>, Author ***@***.***>
Subject: Re: [fmgarage/ft-fms-docker] Linux-ft-fms-docker and hosting a web-page using WebDirect htdocs / httpsRoot (#8)

I just pushed a commit that includes htdocs and did a quick test that worked, please let me know if that works for you.

**Reply by sk-gara** [reply]
Hey, sorry to let you wait, I have been on a vacation.
I've been reading a bit and in the official documentation it says:

> **Note**  For Linux, Custom Web Publishing with PHP and XML is not supported.

[Claris Help](https://help.claris.com/en/server-help/content/hostdb-phps.html)

I'm not sure if it is possible to override that behaviour. Have you done so successfully?

**Comment by ayescas** [comment]
No worries, thank you.  Not sure if installing PHP on the same machine would work.  Definitely can't do it using FMS-Linux.

We use PHP to auto-login into web-direct and pass a script parameter.

Thank you.

From: S ***@***.***>
Date: Friday, August 13, 2021 at 5:04 AM
To: fmgarage/ft-fms-docker ***@***.***>
Cc: Alejandro Yescas ***@***.***>, Author ***@***.***>
Subject: Re: [fmgarage/ft-fms-docker] Linux-ft-fms-docker and hosting a web-page using WebDirect htdocs / httpsRoot (#8)

Hey, sorry to let you wait, I have been on a vacation.
I've been reading a bit and in the official documentation it says:

Note For Linux, Custom Web Publishing with PHP and XML is not supported.

Claris Help<https://help.claris.com/en/server-help/content/hostdb-phps.html>

I'm not sure if it is possible to override that behaviour. Have you done so successfully?
